src/main/scala/com/reagroup/appliedscala/urls/savereview/SaveReviewService.scala

```diff
@@ -15,10 +15,21 @@ class SaveReviewService(saveReview: (MovieId, ValidatedReview) => IO[ReviewId],
     * Return all errors encountered whether the movie exists or not.
     *
     */
-  def save(movieId: MovieId, review: NewReviewRequest): IO[ValidatedNel[ReviewValidationError, ReviewId]] = ???
+  def save(movieId: MovieId, review: NewReviewRequest): IO[ValidatedNel[ReviewValidationError, ReviewId]] = {
```

Comment:
TODO : 
- How did we know this was the data type we wanted to return? `IO[ValidatedNel[ReviewValidationError, ReviewId]]`

- Still really shaky on using `.productR` and `.traverse`

Answer key did it like this:
```scala
  def save(movieId: MovieId, review: NewReviewRequest): IO[ValidatedNel[ReviewValidationError, ReviewId]] = {
    for {
      maybeFetchedMovie <- fetchMovie(movieId)
      maybeValidatedMovie = validateMovie(maybeFetchedMovie)
      maybeValidatedReview = validateReview(review)
      combinedValidations = maybeValidatedMovie.productR(maybeValidatedReview)
      result <- combinedValidations.traverse(validatedReview => saveReview(movieId, validatedReview))
    } yield result
  }
```